api-reference/reference.md

```diff
@@ -1563,6 +1563,480 @@ The intended purpose of the file. Currently, only `batch` is supported.
+  "expires_at": 1744747593
+}
+```
+
+</div>
+</div>
+
+---
+
+### Add upload part
+
+`POST https://api.kluster.ai/v1/uploads/{upload_id}/parts`
+
+Adds a Part to an Upload object. A Part represents a chunk of bytes from the file you are trying to upload.
+
+Each Part can be at most 64 MB, and you can add Parts until you hit the Upload maximum of 8 GB.
+
+It is possible to add multiple Parts in parallel. You can decide the intended order of the Parts when you complete the Upload.
+
+<div class="grid" markdown>
+<div markdown>
+
+**Path parameters**
+
+`upload_id` ++"string"++ <span class="required" markdown>++"required"++</span>
+
+The ID of the Upload.
+
+---
+
+**Request**
+
+`data` ++"file"++ <span class="required" markdown>++"required"++</span>
+
+The chunk of bytes for this Part.
+
+---
+
+**Returns**
+
+The upload Part object.
+
+</div>
+<div markdown>
+
+=== "Python"
+
+    ```python title="Example request"
+    from openai import OpenAI
+
+    # Configure OpenAI client
+    client = OpenAI(
+        base_url="https://api.kluster.ai/v1", 
+        api_key="INSERT_API_KEY" # Replace with your actual API key
+    )
+
+    # Open the file you want to upload and replace upload ID
+    with open("training_examples.jsonl", "rb") as file:
+        part = client.uploads.parts.create(
+            upload_id="INSERT_UPLOAD_ID",
+            data=file
+        )
+
+    print(part.to_dict())
+
+    ```
+
+=== "curl"
+
+    ```bash title="Example request"
+    curl -s https://api.kluster.ai/v1/uploads/INSERT_UPLOAD_ID/parts \
+        -H "Authorization: Bearer $API_KEY" \
+        -F "data=@training_examples.jsonl"
+    ```
+
+```Json title="Response"
+{
+  "id": "67feb0c9f6f1dad39fec8d39",
+  "object": "upload.part",
+  "created_at": 1744744649,
+  "upload_id": "67feae39d63649ef421416f8"
+}
+```
+
+</div>
+</div>
+
+---
+
+### Complete upload
+
+`POST https://api.kluster.ai/v1/uploads/{upload_id}/complete`
+
+Completes the Upload.
+
+Within the returned Upload object, there is a nested File object that is ready to use in the rest of the platform.
+
+You can specify the order of the Parts by passing in an ordered list of the Part IDs.
+
+The number of bytes uploaded upon completion must match the number of bytes initially specified when creating the Upload object. No Parts may be added after an Upload is completed.
+
+<div class="grid" markdown>
+<div markdown>
+
+**Path parameters**
+
+`upload_id` ++"string"++ <span class="required" markdown>++"required"++</span>
+
+The ID of the Upload.
+
+---
+
+**Request**
+
+`part_ids` ++"array"++ <span class="required" markdown>++"required"++</span>
+
+The ordered list of Part IDs.
+
+---
+
+`md5` ++"string"++ <span class="optional" markdown>++"optional"++</span>
+
+The optional md5 checksum for the file contents to verify if the bytes uploaded matches what you expect.
+
+---
+
+**Returns**
+
+The Upload object with status completed with an additional file property containing the created usable File object.
```

Comment:
Could it potentially return other statuses? Like from the [UploadStatus enum](https://github.com/kluster-ai/platform/blob/1ff520935aac776b927b0338556ea30050fe2d7d/common/models/enums/UploadsStatus.ts#L1). If so, we should generalize what it returns instead of saying you'll for sure get a completed status